profiler/external_profiler.py

```diff
@@ -0,0 +1,90 @@
+"""Adding semantic marking to external profiler
+
+Works with nvtx (via cupy) for now.
+"""
+
+import sys
+
+try:
+    import cupy as cp
+except ModuleNotFoundError:
+    cp = None
+
+try:
+    from mpi4py import MPI
+except ModuleNotFoundError:
+    MPI = None
+
+
+def get_stencil_name(frame, event, args) -> str:
+    """Get the name of the stencil from within a call to StencilWrapper.__call__"""
+    name = getattr(
+        frame.f_locals["self"].func, "__name__", repr(frame.f_locals["self"].func)
+    )
+    return f"{name}.__call__"
+
+
+def get_static_name(frame, event, args) -> str:
+    """Static naming"""
+    if (
+        frame.f_code.co_name == "__call__"
+        and frame.f_code.co_filename == "fv3core/stencils/dyn_core.py"
+    ):
+        return "Acoustic timestep"
+    elif (
+        frame.f_code.co_name == "compute"
+        and frame.f_code.co_filename == "fv3core/stencils/remapping.py"
+    ):
+        return "Remapping"
```

Comment:
Might be cool to add TracerAdvections from fv3core/stencils/tracer_2d_1l.py to match the typical plots we show, though not crucial

Reply:
Yep, just had a chat with Tobias I'll add the `TracerAdvection.__call__` and also what I can to mark halo updates